endpoints/api_config.py

```diff
@@ -1062,9 +1062,12 @@ def get_path(self, api_info):
         path = '%s%s%s' % (api_info.path, '/' if path else '', path)
 
     # Verify that the path seems valid.
-    for part in path.split('/'):
+    parts = path.split('/')
+    for n, part in enumerate(parts):
       if part and '{' in part and '}' in part:
-        if re.match('^{[^{}]+}$', part) is None:
+        if (re.match('^{[^{}]+}$', part) is None and
```

Comment:
You should re.compile the regexes and store them as module constants instead, for efficiency's sake. Then I would do something like this:

r = PART_RE if n < len(parts) - 1 else LAST_PART_RE
if not r.match(part):